go/vt/discovery/topology_watcher.go

```diff
@@ -175,7 +180,17 @@ func (tw *TopologyWatcher) loadTablets() {
 		log.Errorf("cannot get tablets for cell: %v: %v", tw.cell, err)
 		return
 	}
-	for _, tAlias := range tabletAlias {
+
+	tw.mu.Lock()
+	for _, tAlias := range tabletAliases {
+		if !tw.refreshKnownTablets {
+			aliasStr := topoproto.TabletAliasString(tAlias)
```

Comment:
Looks like you decided to not use the alias as the key instead. But healthcheck is still using TabletToMapKey. How do the two coordinate correctly?

Reply:
The TopologyWatcher now uses the alias as the key for the internal `tablets` map and all the temporary data structures.

When it calls into healthcheck to add/remove/replace the tablet, it passes the full tablet record. At that point HC recomputes its own hash key from the address map. I think we could (and probably should) switch that to store tablets keyed by the alias as well, but it's not necessary as part of this change.